grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -704,16 +703,22 @@ class ShipmentService {
                 throw new ValidationException("Shipment item is invalid", shipmentItem.errors)
             }
 
-            // Check whether there's any stock in the bin location for the given inventory item
-            def quantityOnHand = getQuantityOnHand(origin, shipmentItem.binLocation, shipmentItem.inventoryItem, binLocationRequired)
+            def quantityAvailableToPromise = productAvailabilityService.getQuantityAvailableToPromise(origin, shipmentItem.binLocation, shipmentItem.inventoryItem)
+            def quantityPicked
+            if (shipmentItem?.binLocation) {
```

Comment:
Is it possible to clean this up? Maybe move it to the shipmentItem
```
def quantityPicked = shipmentItem.quantityPicked
```

Reply:
where quantityPicked is a getter method getQuantityPicked that'll look something like this
```
def getQuantityPicked() { 
    if (binLocation) { 
        return requisitionItem?.picklistItems?.findAll { it.inventoryItem == shipmentItem?.inventoryItem && it.binLocation == shipmentItem?.binLocation }?.sum { it.quantity }
    } else {
        return shipmentItem?.requisitionItem?.picklistItems?.findAll { it.inventoryItem == shipmentItem?.inventoryItem }?.sum { it.quantity }
    }
}
```